diff --git a/Makefile b/Makefile
index b2476db..b9d9dac 100644
--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,11 @@ CFLAGS_LIB = $(CFLAGS_FAST) -fPIC
 
 LDFLAGS_LIB = $(LDFLAGS) -shared
 
+ifeq (darwin,$(PLATFORM))
 INSTALL ?= install
+else
+INSTALL ?= install -D
+endif
 PREFIX ?= $(DESTDIR)/usr/local
 LIBDIR = $(PREFIX)/lib
 INCLUDEDIR = $(PREFIX)/include
@@ -123,15 +127,22 @@ tags: http_parser.c http_parser.h test.c
 	ctags $^
 
 install: library
-	$(INSTALL) -D  http_parser.h $(INCLUDEDIR)/http_parser.h
-	$(INSTALL) -D $(SONAME) $(LIBDIR)/$(SONAME)
+	$(INSTALL) -d $(INCLUDEDIR) $(LIBDIR)
+	$(INSTALL) http_parser.h $(INCLUDEDIR)/http_parser.h
+	$(INSTALL) $(SONAME) $(LIBDIR)/$(SONAME)
 	ln -s $(LIBDIR)/$(SONAME) $(LIBDIR)/libhttp_parser.$(SOEXT)
 
 install-strip: library
-	$(INSTALL) -D  http_parser.h $(INCLUDEDIR)/http_parser.h
-	$(INSTALL) -D -s $(SONAME) $(LIBDIR)/$(SONAME)
+	$(INSTALL) -d $(INCLUDEDIR) $(LIBDIR)
+	$(INSTALL) http_parser.h $(INCLUDEDIR)/http_parser.h
+	$(INSTALL) -s $(SONAME) $(LIBDIR)/$(SONAME)
 	ln -s $(LIBDIR)/$(SONAME) $(LIBDIR)/libhttp_parser.$(SOEXT)
 
+install-static: package
+	$(INSTALL) -d $(INCLUDEDIR) $(LIBDIR)
+	$(INSTALL) http_parser.h $(INCLUDEDIR)/http_parser.h
+	$(INSTALL) libhttp_parser.a $(LIBDIR)/libhttp_parser.a
+
 uninstall:
 	rm $(INCLUDEDIR)/http_parser.h
 	rm $(LIBDIR)/$(SONAME)
@@ -146,4 +157,4 @@ clean:
 contrib/url_parser.c:	http_parser.h
 contrib/parsertrace.c:	http_parser.h
 
-.PHONY: clean package test-run test-run-timed test-valgrind install install-strip uninstall
+.PHONY: clean package test-run test-run-timed test-valgrind install install-strip install-static uninstall
